enhancements/ingress/gcp-lb-global-access.md

```diff
@@ -0,0 +1,223 @@
+is used heavily.
+
+Additionally, cluster administrators need to ensure that enabling the Global Access option does not over-expose an
+Ingress Controller's front end within the given VPC.
+
+## Design Details
+
+### Open Questions
+
+How often do cluster administrators want an internal load balancer on GCP that can communicate with clients in a different region?
+
+### Test Plan
+
+The Ingress Operator has existing unit tests to verify that the operator renders the correct service definition given an Ingress Controller's `Spec.LoadBalancer` field.
+These unit test will be extended to verify that Ingress Operator applies the Global Access LoadBalancer service annotations for Ingress Controllers with
+`Spec.LoadBalancer.ProviderParameters.GCP.ClientAccess` and `Spec.LoadBalancer.Scope` properly set.
+
+The Ingress Operator also has existing e2e tests. These e2e tests will be extended to verify that Ingress Controller's created with the Global Access option enabled
+successfully create a LoadBalancer service and function as intended. Note that any e2e test created for this enhancement would only run on GCP clusters. With that in mind,
+additional CI jobs may need to be added to the Ingress Operator to accommodate this need.
+
+
+An example of a proper e2e test is as follows:
+
+1. Create an Ingress Controller with an Internal LoadBalancer and the `ClientAccess` field set to `Global`.
+2. Ensure that the LoadBalancer service is created.
+3. Ensure that the Ingress Controller works normally.
+4. Modify the Ingress Controller's spec so that the `ClientAccess` field is set to `Local`.
+5. Ensure that the LoadBalancer service is properly updated with the correct annotations.
+
+Note that testing the actual behavior of the Global Access option from within GCP by using a client in a different region is most likely
+not feasible given CI limitations, and most likely unnecessary. In other words, it is most likely safe to assume that GCP Internal Load Balancers
+set with the proper Global Access annotations work as described by the [GCP documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/internal-load-balancing#global_access).
+
+### Graduation Criteria
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, internal LoadBalancer services on GCP will not be updated with the Global Access annotation unless
+an Ingress Controller's spec is updated to require so.
+
+On downgrade, the Ingress Operator will ignore the new `GCPLoadBalancerParameters` field for an Ingress Controller and not set any
+Global Access LoadBalancer service annotations.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+None.
+
+## Alternatives
+
+A cluster administrator is already free to use a load balancer not managed by the Cluster Ingress Operator to configure these kinds of settings on an Ingress Controller's
+LoadBalancer service.
```

Comment:
This could be clearer:
```suggestion
A cluster administrator can already create a custom service that is not managed by the Cluster Ingress Operator.  The administrator can then configure these kinds of settings on the
custom, administrator-managed LoadBalancer service.
```

Reply:
Thanks for the suggestion! fixed.